src/wallet/components/WalletDropdownFundLink.tsx

```diff
@@ -0,0 +1,76 @@
+import { isValidElement, useMemo } from 'react';
+import { FundWalletSvg } from '../../internal/svg/fundWallet';
+import { cn, pressable, text as themeText } from '../../styles/theme';
+import { version } from '../../version';
+import type { WalletDropdownFundLinkReact } from '../types';
+import { getWindowDimensions } from '../utils/getWindowDimensions';
+
+export function WalletDropdownFundLink({
+  className,
+  icon,
+  rel,
+  popupFeatures,
+  openIn = 'tab',
+  target,
+  text = 'Fund wallet',
+  popupSize = 'md',
+}: WalletDropdownFundLinkReact) {
+  const iconSvg = useMemo(() => {
+    if (icon === undefined) {
+      return FundWalletSvg;
+    }
+    if (isValidElement(icon)) {
+      return icon;
+    }
+  }, [icon]);
+
+  const currentURL = window.location.href;
+  const tabName = document.title;
+  const fundingUrl = `http://keys.coinbase.com/fund?dappName=${tabName}&dappUrl=${currentURL}&version=${version}&source=onchainkit`;
+
+  const handleClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    const { width, height } = getWindowDimensions(popupSize);
+
+    const left = Math.round((window.screen.width - width) / 2);
+    const top = Math.round((window.screen.height - height) / 2);
+
+    const windowFeatures =
+      popupFeatures ||
+      `width=${width},height=${height},resizable,scrollbars=yes,status=1,left=${left},top=${top}`;
+    window.open(fundingUrl, target, windowFeatures);
+  };
+
+  const overrideClassName = cn(
+    pressable.default,
+    'relative flex items-center px-4 py-3',
+    className,
+  );
+
+  const linkContent = (
```

Comment:
you could wrap this in useMemo. benefit would be small but i think it is best practice.